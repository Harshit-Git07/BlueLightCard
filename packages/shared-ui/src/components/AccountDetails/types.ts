export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
