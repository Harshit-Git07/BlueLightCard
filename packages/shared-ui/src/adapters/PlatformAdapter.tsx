import { createContext, useContext } from 'react';
import { AmplitudeLogParams, PlatformVariant } from '../types';

export type HttpMethod = 'GET' | 'PUT' | 'POST' | 'DELETE' | 'PATCH' | 'HEAD' | 'OPTIONS' | 'TRACE';

export type V5RequestOptions = {
  /**
   * Platforms should not cache responses if the cachePolicy is set to 'never'.
   * Setting the cachePolicy to 'auto' allows the platform to cache responses but
   * does not require it to do so. If this option is omitted, platforms should
   * default to 'never'.
   *
   * @default 'never'
   */
  cachePolicy?: 'never' | 'auto';
  method: HttpMethod;
  queryParameters?: Record<string, string>;
  body?: string;
};

export type V5Response = {
  statusCode: number;
  body: string;
};

export interface IPlatformAdapter {
  /**
   * Can be used by the shared-ui to introspect which platform it is running in
   */
  platform: PlatformVariant;
  /**
   * Invokes a v5 API endpoint
   */
  invokeV5Api(path: string, options: V5RequestOptions): Promise<V5Response>;
  /**
   * Logs analytics events
   */
  logAnalyticsEvent(event: string, parameters: AmplitudeLogParams): void;
  /**
   * Navigate to a route within the app
   */
  navigate(path: string): void;
  /**
   * Navigate to an external URL
   */
  navigateExternal(url: string): void;
}

const PlatfromAdapterContext = createContext<IPlatformAdapter | null>(null);

type Props = {
  children: React.ReactNode;
  adapter: IPlatformAdapter;
};

export function PlatformAdapterProvider({ adapter, children }: Props): React.ReactNode {
  return (
    <PlatfromAdapterContext.Provider value={adapter}>{children}</PlatfromAdapterContext.Provider>
  );
}

export function usePlatformAdapter() {
  const adapter = useContext(PlatfromAdapterContext);

  if (!adapter) {
    throw new Error(
      'Missing platform adapter implementation. Did you forget to wrap your app with <PlatformAdapterProvider>?',
    );
  }

  return adapter;
}
