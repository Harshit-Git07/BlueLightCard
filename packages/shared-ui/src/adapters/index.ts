export * from './PlatformAdapter';
export * from './useMockPlatformAdapter';
export * from './StorybookPlatformAdapter';
