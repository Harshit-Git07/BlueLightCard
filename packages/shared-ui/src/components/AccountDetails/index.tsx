import { ThemeVariant, Typography } from '@bluelightcard/shared-ui';
import CopyButton from '../CopyButton/CopyButton';
import { AccountDetailsProps } from './types';

const AccountDetails = ({ accountNumber, firstName, lastName }: AccountDetailsProps) => {
  const greetings = `Hi ${firstName} ${lastName},`;

  return (
    <div className={`flex flex-col`}>
      <Typography className="hidden tablet:block" variant="headline">
        {greetings}
      </Typography>

      <Typography className="block tablet:hidden" variant="title-large">
        {greetings}
      </Typography>

      <div className="mt-2 flex items-center gap-2">
        <Typography
          className="text-colour-onSurface-light dark:text-colour-onSurface-dark"
          variant="label-semibold"
        >
          Card number
        </Typography>

        <Typography
          className="text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark"
          variant="body-light"
        >
          {accountNumber}
        </Typography>

        <CopyButton
          variant={ThemeVariant.Tertiary}
          label="Copy"
          copyText={accountNumber}
          className="ml-1"
        />
      </div>
    </div>
  );
};

export default AccountDetails;
