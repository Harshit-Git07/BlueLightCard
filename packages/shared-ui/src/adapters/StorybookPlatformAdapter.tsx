import { Decorator } from '@storybook/react';
import { IPlatformAdapter, PlatformAdapterProvider } from './PlatformAdapter';
import { PlatformVariant } from '../types';

export const storybookPlatformAdapter: IPlatformAdapter = {
  getAmplitudeFeatureFlag: () => 'control',
  invokeV5Api: () => Promise.resolve({ status: 200, data: '{}' }),
  logAnalyticsEvent: () => {},
  navigate: () => {},
  navigateExternal: () => ({
    isOpen: () => true,
  }),
  writeTextToClipboard: () => Promise.resolve(),
  getBrandURL: () => '',
  platform: PlatformVariant.MobileHybrid,
} satisfies IPlatformAdapter;

export const StorybookPlatformAdapterDecorator: Decorator = (Story, { parameters }) => {
  const platformAdapter = {
    ...storybookPlatformAdapter,
    ...parameters.platformAdapter,
  };

  return (
    <PlatformAdapterProvider adapter={platformAdapter}>
      <Story platformAdapter={platformAdapter} />
    </PlatformAdapterProvider>
  );
};
