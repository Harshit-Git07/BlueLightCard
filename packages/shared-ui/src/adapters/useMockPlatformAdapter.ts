import { IPlatformAdapter } from './PlatformAdapter';
import { PlatformVariant } from '../types';

export const useMockPlatformAdapter = (
  status = 200,
  data = {},
  platform = PlatformVariant.MobileHybrid,
) => {
  const getAmplitudeFeatureFlag = jest.fn();
  const invokeV5Api = jest.fn().mockResolvedValue({ status, data: JSON.stringify(data) });
  const logAnalyticsEvent = jest.fn();
  const navigate = jest.fn();
  const navigateExternal = jest.fn();
  const writeTextToClipboard = jest.fn().mockReturnValue(Promise.resolve());

  const mockPlatformAdapter = {
    platform,
    getAmplitudeFeatureFlag,
    invokeV5Api,
    logAnalyticsEvent,
    navigate,
    navigateExternal,
    writeTextToClipboard,
  } satisfies IPlatformAdapter;

  return mockPlatformAdapter;
};
