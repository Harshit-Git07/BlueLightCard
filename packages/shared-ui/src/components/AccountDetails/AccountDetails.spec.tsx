import { PlatformAdapterProvider, useMockPlatformAdapter } from '../../adapters';
import renderer from 'react-test-renderer';
import AccountDetails from './index';

describe('AccountDetails', () => {
  const mockPlatformAdapter = useMockPlatformAdapter();

  it('should render AccountDetails correctly', () => {
    const props = {
      alertType: 'success',
      title: 'Success Alert',
      description: 'This is a success message.',
    };

    const component = renderer.create(
      <PlatformAdapterProvider adapter={mockPlatformAdapter}>
        <AccountDetails accountNumber="BLC0000000" firstName="Name" lastName="Last-name" />
      </PlatformAdapterProvider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
