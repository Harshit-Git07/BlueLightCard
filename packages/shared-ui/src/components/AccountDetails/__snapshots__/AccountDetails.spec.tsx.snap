// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails should render AccountDetails correctly 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="text-heading-colour dark:text-heading-colour-dark font-typography-headline font-typography-headline-weight text-typography-headline leading-typography-headline tracking-typography-headline hidden tablet:block"
  >
    Hi Name Last-name,
  </div>
  <div
    className="text-heading-colour dark:text-heading-colour-dark font-typography-title-large font-typography-title-large-weight text-typography-title-large leading-typography-title-large tracking-typography-title-large block tablet:hidden"
  >
    Hi Name Last-name,
  </div>
  <div
    className="mt-2 flex items-center gap-2"
  >
    <div
      className="text-heading-colour dark:text-heading-colour-dark font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold tracking-typography-label-semibold text-colour-onSurface-light dark:text-colour-onSurface-dark"
    >
      Card number
    </div>
    <div
      className="text-heading-colour dark:text-heading-colour-dark font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light tracking-typography-body-light text-colour-onSurface-subtle-light dark:text-colour-onSurface-subtle-dark"
    >
      BLC0000000
    </div>
    <div
      className="ml-1 flex items-center"
    >
      <button
        aria-label="copy"
        className="font-typography-label-semibold font-typography-label-semibold-weight text-typography-label-semibold leading-typography-label-semibold  text-button-tertiary-default-label-colour-light dark:text-button-tertiary-default-label-colour-dark hover:text-button-tertiary-hover-label-colour-light dark:hover:text-button-tertiary-hover-label-colour-dark  border-none py-0   text-button-label-font font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
        name="copy"
        onClick={[Function]}
        type="button"
      >
        Copy
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-copy ml-2"
          data-icon="copy"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M208 0H332.1c12.7 0 24.9 5.1 33.9 14.1l67.9 67.9c9 9 14.1 21.2 14.1 33.9V336c0 26.5-21.5 48-48 48H208c-26.5 0-48-21.5-48-48V48c0-26.5 21.5-48 48-48zM48 128h80v64H64V448H256V416h64v48c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V176c0-26.5 21.5-48 48-48z"
            fill="currentColor"
            style={{}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
